### PR TITLE
19460: Coverage_selected status with enrollees who have outstanding verifications

### PR DESCRIPTION
**Pull request by hitheshkolla**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [19460](https://devops.dchbx.org/redmine/issues/19460) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | 12/31/2017  |
| App-Dev Priority | UP-1 |
| Development Story Points | n/a |
| Peer Review Story Points | 0.5 |
| Ticket Author | Sarah Bagge |

### CODE REVIEW

File: app/data_migrations/fix_hub_verified_consumer.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def update_verification_type(person, v_type)
                                                    } )
     end
 
-    if person.all_types_verified?
+    if person.all_types_verified? && !person.consumer_role.fully_verified?
```
Comment (saikumar9) [inline app/data_migrations/fix_hub_verified_consumer.rb]: Additional check is added to verify if a person is **Fully Verified.**

File: app/data_migrations/update_verification_types.rb [hunk 2]
```diff
@@ -2,17 +2,23 @@
...
-        person.consumer_role.update_all_verification_types unless person.consumer_role.all_types_verified?
+        update_local_residency(person, "datamigration")
+        puts "#{i} Updating... #{person.id}" unless Rails.env.test?
       rescue
         $stderr.puts "Issue migrating person: #{person.fullname}, #{person.hbx_id}, #{person.id}" unless Rails.env.test?
       end
     end
   end
+
+  def update_local_residency(person, update_reason)
```
Comment (saikumar9) [inline app/data_migrations/update_verification_types.rb]: Consumer role's local residency is updated with a valid reason.

File: spec/data_migrations/update_verification_types_spec.rb [hunk 3]
```diff
@@ -2,37 +2,39 @@
...
 
   context "fully verified person" do
-    it_behaves_like "update verification types for fully verified people", true, "hbx", "hbx"
-    it_behaves_like "update verification types for fully verified people", true, "curam", "curam"
-    it_behaves_like "update verification types for fully verified people", true, "any", "hbx"
+    it_behaves_like "update residency type for verified people", "fully_verified", nil, "is_state_resident", true
+    it_behaves_like "update residency type for verified people", "fully_verified", nil, "local_residency_validation", "valid"
+  end
+
+  context "sci verified person" do
```
Comment (saikumar9) [inline spec/data_migrations/update_verification_types_spec.rb]: Good test coverage.

**Comment by Kevin-Wei** [comment]
extra commit 76023ef79a4bf3f38a5d9dd46a155598219d1d85 will be refactored

**Pull request closed without merging**